x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/actions/info_button.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiDescriptionList,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiPortal,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import moment from 'moment';
+import React, { Component, Fragment } from 'react';
+import { Filter, Query, TimeRange } from 'src/plugins/data/public';
+import { UISession } from '../../types';
+import { TableText } from '../';
+
+interface Props {
+  searchSession: UISession;
+}
+
+interface State {
+  isLoading: boolean;
+  isFlyoutVisible: boolean;
+  calloutTitle: string;
+  error: Error | null;
+}
+
+export class InfoButton extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+      isFlyoutVisible: false,
+      calloutTitle: 'Search Session Info',
+      error: null,
+    };
+
+    this.closeFlyout = this.closeFlyout.bind(this);
+    this.showFlyout = this.showFlyout.bind(this);
+  }
+
+  public renderInfo() {
+    const { error: err } = this.state;
+    if (err) {
+      return err.message;
+    }
+
+    const { created, id, initialState, restoreState } = this.props.searchSession;
+
+    const sessionInfo = [
+      {
+        title: 'Search Session Id',
+        description: id,
+      },
+      {
+        title: 'Created at',
+        description: moment(created).toLocaleString(),
+      },
+    ];
+
+    if (initialState.timeRange) {
```

Comment:
Should we just render initialState / restoreState as a raw JSON instead? 
Why I suggest this: 

1. For debug/qa purposes it will still do the job and would give even more info
2. If we'd want to display anything else, we'd not need to get back and modify this file
3. *Current code relies on `timeRange`, `query`, `filters` in the state object, but it is app-specific and could be different for other apps: missing or named differently or be nested somewhere*. 
4. This also isn't typed.

Reply:
I showed just `initialState` in a `CodeEditor` component.
Thanks for the idea!